data-prep/encode-answers-code.py

```diff
@@ -0,0 +1,23 @@
+#%%
+import torch
+from torch.nn import CosineSimilarity
+from transformers import pipeline
+
+
+def embeddings(text):
+    feature_extraction = pipeline(
+        "feature-extraction",
+        model="microsoft/codebert-base",
+        tokenizer="microsoft/codebert-base",
+    )
+    features = feature_extraction(text)
+    return features
+
+
+s1 = embeddings("def greet(user):\n print(f'hello {user}!')")
+s2 = embeddings("def greet(person):\n print(f'hello {person}!')")
+s3 = embeddings("x")
+
+
+cos = CosineSimilarity()
```

Comment:
@szemyd I'm receiving this error, can you take a second look?
```
TypeError                                 Traceback (most recent call last)
Cell In [65], line 23
     19 s3 = embeddings("x")
     22 cos = CosineSimilarity()
---> 23 print(cos(s1, s3))

File ~/anaconda3/envs/ncs-data-prep/lib/python3.9/site-packages/torch/nn/modules/module.py:1102, in Module._call_impl(self, *input, **kwargs)
   [1098](file:///Users/zoltanctoth/anaconda3/envs/ncs-data-prep/lib/python3.9/site-packages/torch/nn/modules/module.py?line=1097) # If we don't have any hooks, we want to skip the rest of the logic in
   [1099](file:///Users/zoltanctoth/anaconda3/envs/ncs-data-prep/lib/python3.9/site-packages/torch/nn/modules/module.py?line=1098) # this function, and just call forward.
   [1100](file:///Users/zoltanctoth/anaconda3/envs/ncs-data-prep/lib/python3.9/site-packages/torch/nn/modules/module.py?line=1099) if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks or _global_backward_hooks
   [1101](file:///Users/zoltanctoth/anaconda3/envs/ncs-data-prep/lib/python3.9/site-packages/torch/nn/modules/module.py?line=1100)         or _global_forward_hooks or _global_forward_pre_hooks):
-> [1102](file:///Users/zoltanctoth/anaconda3/envs/ncs-data-prep/lib/python3.9/site-packages/torch/nn/modules/module.py?line=1101)     return forward_call(*input, **kwargs)
   [1103](file:///Users/zoltanctoth/anaconda3/envs/ncs-data-prep/lib/python3.9/site-packages/torch/nn/modules/module.py?line=1102) # Do not call functions when jit is used
   [1104](file:///Users/zoltanctoth/anaconda3/envs/ncs-data-prep/lib/python3.9/site-packages/torch/nn/modules/module.py?line=1103) full_backward_hooks, non_full_backward_hooks = [], []

File ~/anaconda3/envs/ncs-data-prep/lib/python3.9/site-packages/torch/nn/modules/distance.py:77, in CosineSimilarity.forward(self, x1, x2)
     [76](file:///Users/zoltanctoth/anaconda3/envs/ncs-data-prep/lib/python3.9/site-packages/torch/nn/modules/distance.py?line=75) def forward(self, x1: Tensor, x2: Tensor) -> Tensor:
---> [77](file:///Users/zoltanctoth/anaconda3/envs/ncs-data-prep/lib/python3.9/site-packages/torch/nn/modules/distance.py?line=76)     return F.cosine_similarity(x1, x2, self.dim, self.eps)

TypeError: cosine_similarity(): argument 'x1' (position 1) must be Tensor, not list
```

Reply:
- I think the problem is that we have a list instead of a tensor at S3, so we could simply cast it as such: ``torch.Tensor(list)``
However we need to create a single vector out of all the vectors that the embedding process creates, becuase it does the embedding by word and not by sentence/document/codesnippet.